src/role_atkeeper.js

```diff
@@ -9,17 +9,14 @@
 
 roles.atkeeper = {};
 
-roles.atkeeper.getPartConfig = function(room, energy, heal) {
-  var parts = [MOVE, RANGED_ATTACK, MOVE, HEAL];
-  return room.getPartConfig(energy, parts);
-};
-
-roles.atkeeper.energyRequired = function(room) {
-  return Math.min(room.getEnergyCapacityAvailable(), 3430);
-};
-
-roles.atkeeper.energyBuild = function(room) {
-  return Math.min(room.getEnergyCapacityAvailable(), 3430);
+roles.atkeeper.settings = {
+  parts: {
+    layout: 'MMRH'
+  },
+  energy: {
+    minEnergyStored: 3430,
+    maxEnergyUsed: 3430
+  }
```

Comment:
dunno... About maxEnergyUsed it was often use for have no more than 50 parts on actual master branch. i can surely already remove them but must parse all before ^^'
I thought also about use min/max repeat instead of setup min/max energy as i'll can setup defaults values as 1 / null where null is the maximum repeat possible.
About parts, i can maybe setup some default paterns in config as Worker : 'MW', hauler : 'M', healer : 'MH' but it will be pretty useless as a worker will quickly can upgrade to 'MWW...'